### PR TITLE
ES-789

### PR DESCRIPTION
**Pull request by mengleang-0090**
https://mosip.atlassian.net/browse/ES-789

### CODE REVIEW

**Review by rathanak-0080 — APPROVED** [review]

File: signup-service/src/test/java/io/mosip/signup/controllers/RegistrationControllerTest.java
Comment (ase-101) [inline signup-service/src/test/java/io/mosip/signup/controllers/RegistrationControllerTest.java:1]: @mengleang-0090 Do we have a test case to check if the locale is a null/empty string or an invalid language code?

File: signup-service/src/main/java/io/mosip/signup/dto/RegisterRequest.java [hunk 1]
```diff
@@ -25,4 +25,6 @@ public class RegisterRequest {
 
     @NotNull(message = ErrorConstants.INVALID_USERINFO)
     private @Valid UserInfoMap userInfo;
+
+    private String locale;
```
Comment (ase-101) [inline signup-service/src/main/java/io/mosip/signup/dto/RegisterRequest.java:29]: Do we need validation only if a value is provided in this field? I believe we have language validator annotation already with a required boolean flag @mengleang-0090

File: signup-service/src/main/java/io/mosip/signup/dto/ResetPasswordRequest.java [hunk 2]
```diff
@@ -12,4 +12,6 @@ public class ResetPasswordRequest {
 
     @Password
     private String password;
+
+    private String locale;
```
Comment (ase-101) [inline signup-service/src/main/java/io/mosip/signup/dto/ResetPasswordRequest.java:16]: Do we need validation only if a value is provided in this field? I believe we have language validator annotation already with a required boolean flag

File: signup-ui/src/pages/shared/mutations.ts [hunk 3]
```diff
@@ -55,29 +56,37 @@ export const useVerifyChallenge = () => {
...
+      return register(registrationRequestDto)},
     gcTime: Infinity,
   });
 
   return { registerMutation };
 };
 
 export const useResetPassword = () => {
+  const { i18n } = useTranslation();
+  const locale = i18n.language == "km" ? "khm" : "eng";
```
Comment (ase-101) [inline signup-ui/src/pages/shared/mutations.ts:79]: covnerting from 2 letter language code to 3 letter for backend service cannot be handled with a library? We should have solved similar problem in eSignet UI. Can you please connect with @aranaravi for this @mengleang-0090.

File: signup-ui/src/pages/shared/mutations.ts [hunk 4]
```diff
@@ -55,29 +56,37 @@ export const useVerifyChallenge = () => {
 };
 
 export const useRegister = () => {
+  const { i18n } = useTranslation();
+  const locale = i18n.language == "km" ? "khm" : "eng";
```
Comment (ase-101) [inline signup-ui/src/pages/shared/mutations.ts:60]: covnerting from 2 letter language code to 3 letter for backend service cannot be handled with a library? We should have solved similar problem in eSignet UI. Can you please connect with @aranaravi for this @mengleang-0090.

File: signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java [hunk 5]
```diff
@@ -324,7 +324,7 @@ public RegistrationStatusResponse updatePassword(ResetPasswordRequest resetPassw
         transaction.setRegistrationStatus(RegistrationStatus.PENDING);
         cacheUtilService.setStatusCheckTransaction(transactionId, transaction);
 
-        notificationHelper.sendSMSNotificationAsync(resetPasswordRequest.getIdentifier(), transaction.getLocale(),
+        notificationHelper.sendSMSNotificationAsync(resetPasswordRequest.getIdentifier(), resetPasswordRequest.getLocale(),
```
Comment (ase-101) [inline signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java:327]: if the language code is null in the request, we can take the language code from the transaction. what do you think @mengleang-0090?

File: signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java [hunk 6]
```diff
@@ -251,7 +251,7 @@ public RegisterResponse register(RegisterRequest registerRequest, String transac
         transaction.setRegistrationStatus(RegistrationStatus.PENDING);
         cacheUtilService.setStatusCheckTransaction(transactionId, transaction);
 
-        notificationHelper.sendSMSNotificationAsync(registerRequest.getUserInfo().getPhone(), transaction.getLocale(),
+        notificationHelper.sendSMSNotificationAsync(registerRequest.getUserInfo().getPhone(), registerRequest.getLocale(),
```
Comment (ase-101) [inline signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java:254]: if the language code is null in the request, we can take the language code from the transaction. what do you think @mengleang-0090?

File: signup-service/src/test/java/io/mosip/signup/controllers/RegistrationControllerTest.java [hunk 7]
```diff
@@ -1057,9 +1063,10 @@ public void register_withValidFullName_returnSuccessResponse() throws Exception{
 
         RegisterResponse registerResponse = new RegisterResponse();
         registerResponse.setStatus(ActionStatus.PENDING);
-        when(registrationService.register(registerRequest, mockTransactionID)).thenReturn(registerResponse);
+        when(registrationService.register(eq(registerRequest), eq(mockTransactionID))).thenReturn(registerResponse);
 
         mockMvc.perform(post("/registration/register")
+                        .header("locale", "khm")
```
Comment (ase-101) [inline signup-service/src/test/java/io/mosip/signup/controllers/RegistrationControllerTest.java:1069]: Do we still need locale in the header?

**Review by ase-101 — APPROVED** [review]

**Pull request merged**